lib/ansible/playbook/role/__init__.py

```diff
@@ -39,6 +39,11 @@
 from ansible.utils.path import is_subpath
 from ansible.utils.vars import combine_vars
 
+# global cache for role dependencies
+# Key: collection.path.role_name
+# Value: list of role dependencies
+global_role_dependency_cache : dict[str, list[str]] = {}
```

Comment:
The linters will probably be unhappy about that whitespace before colon.
```suggestion
global_role_dependency_cache: dict[str, list[str]] = {}
```